assets/javascripts/discourse/services/chat-draft-handler.js

```diff
@@ -0,0 +1,49 @@
+import Service from "@ember/service";
+import discourseDebounce from "discourse-common/lib/debounce";
+import { ajax } from "discourse/lib/ajax";
+
+export default class ChatDraftHandler extends Service {
+  init() {
+    super.init(...arguments);
+
+    this.draftStore = {};
+    if (this.currentUser.chat_drafts) {
+      this.currentUser.chat_drafts.forEach((draft) => {
+        this.draftStore[draft.channel_id] = JSON.parse(draft.data);
+      });
+    }
+  }
+
+  _saveDraft(channelId, draft) {
+    const data = { channel_id: channelId };
+    if (draft) {
+      data.data = JSON.stringify(draft);
+    }
+
+    ajax("/chat/drafts", { type: "POST", data });
+  }
+
+  setDraftForChannel(channel, draft) {
```

Comment:
we don't need to have draft in the name of the function here, given we are already in draft service, same of others

Reply:
Done